test/cypress/integration/payment-links/amount-hint.cy.test.js

```diff
@@ -0,0 +1,143 @@
+const userStubs = require('../../stubs/user-stubs')
+const gatewayAccountStubs = require('../../stubs/gateway-account-stubs')
+const tokenStubs = require('../../stubs/token-stubs')
+const productStubs = require('../../stubs/products-stubs')
+const userExternalId = 'a-user-id'
+const gatewayAccountId = 42
+const gatewayAccountExternalId = 'a-valid-account-id'
+const serviceExternalId = 'a-valid-service-id'
+const serviceName = {
+  en: 'pay for something',
+  cy: 'talu am rywbeth'
+}
+const productName = 'Test'
+const amountHint = 'A hint'
+
+describe('A payment link with an amount hint set', () => {
```

Comment:
Do we need a test where the amount hint is not shown if the env var PAYMENT_LINKS_FUTURE_ENABLED is false / not set?

Reply:
Ideally, yes - but I haven't found a way to change the value of environment variables between different cypress tests.

So this will be a little bit of a blind spot for testing. But as this is selfservice we can probably get away without everything being tested in a super strict way. They worst that could happen if this does show up is the user enters a hint that then isn't currently used by anything, but will be in the future.